crates/chainio/txmanager/src/simple_tx_manager.rs

```diff
@@ -0,0 +1,265 @@
+use alloy_consensus::{TxEip1559, TxLegacy};
+use alloy_network::{Ethereum, TxSigner};
+use alloy_primitives::Address;
+use alloy_provider::{PendingTransactionBuilder, Provider, ProviderBuilder, RootProvider};
+use alloy_rpc_types_eth::TransactionReceipt;
+use alloy_signer_local::PrivateKeySigner;
+use eigen_logging::{logger::Logger, tracing_logger::TracingLogger};
+use eigen_signer::signer::Config;
+use k256::ecdsa::SigningKey;
+use reqwest::Url;
+use thiserror::Error;
+
+pub type Transport = alloy_transport_http::Http<reqwest::Client>;
+
+/// Possible errors raised in Tx Manager
+#[derive(Error, Debug)]
+pub enum TxManagerError {
+    #[error("signer error")]
+    SignerError,
+    #[error("send error")]
+    SendTxError,
+    #[error("wait_for_receipt error")]
+    WaitForReceiptError,
+    #[error("address error")]
+    AddressError,
+    #[error("invalid url error")]
+    InvalidUrlError,
+}
+
+pub struct SimpleTxManager<'log> {
+    logger: &'log TracingLogger,
+    gas_limit_multiplier: f64,
+    private_key: String,
+    provider: RootProvider<Transport>,
+}
+
+impl<'log> SimpleTxManager<'log> {
+    /// Creates a new SimpleTxManager.
+    ///
+    /// # Arguments
+    ///
+    /// - `logger`: The logger to be used.
+    /// - `gas_limit_multiplier`: The gas limit multiplier.
+    /// - `private_key`: The private key of the wallet.
+    /// - `rpc_url`: The RPC URL. It could be an anvil node or any other node.
+    ///
+    /// # Returns
+    ///
+    /// - The SimpleTxManager created.
+    ///
+    /// # Errors
+    ///
+    /// - If the URL is invalid.
+    pub fn new(
+        logger: &'log TracingLogger,
+        gas_limit_multiplier: f64,
+        private_key: &str,
+        rpc_url: &str,
+    ) -> Result<SimpleTxManager<'log>, TxManagerError> {
+        let url = Url::parse(rpc_url).map_err(|_| TxManagerError::InvalidUrlError)?;
```

Comment:
can you add some logging here and elsewhere? errors in particular, but successes too if the go sdk also does it

Reply:
this is done